crates/starknet-server/src/api/json_rpc/endpoints.rs

```diff
@@ -360,3 +384,76 @@ impl JsonRpcHandler {
         Ok(nonce)
     }
 }
+
+fn convert_broadcasted_tx(
+    broadcasted_tx: BroadcastedTransaction,
+    chain_id: StarknetChainId,
+) -> RpcResult<starknet_in_rust_tx::Transaction> {
+    match broadcasted_tx {
+        BroadcastedTransaction::Invoke(BroadcastedInvokeTransaction::V0(_)) => {
+            Err(ApiError::UnsupportedAction { msg: "Invoke V0 is not supported".into() })
+        }
+        BroadcastedTransaction::Invoke(BroadcastedInvokeTransaction::V1(broadcasted_tx)) => {
+            let selector =
+                Felt::from(starknet_rs_core::utils::get_selector_from_name("__execute__").unwrap())
+                    .into();
+
+            Ok(starknet_in_rust_tx::Transaction::InvokeFunction(
+                starknet_in_rust_tx::InvokeFunction::new(
+                    broadcasted_tx.sender_address.0.try_into()?,
+                    selector,
+                    broadcasted_tx.common.max_fee.0,
+                    broadcasted_tx.common.version.into(),
+                    broadcasted_tx.calldata.iter().map(|s| s.into()).collect(),
+                    broadcasted_tx.common.signature.iter().map(|s| s.into()).collect(),
+                    chain_id.to_felt(),
+                    Some(broadcasted_tx.common.nonce.into()),
+                )?,
+            ))
+        }
+        BroadcastedTransaction::Declare(BroadcastedDeclareTransaction::V1(broadcasted_tx)) => {
+            let contract_class: starknet_types::contract_class::ContractClass = broadcasted_tx
+                .contract_class
+                .try_into() // TODO why does this work if the converter is in write_endpoints.rs?
```

Comment:
because the trait TryFrom is implemented on the DeprecatedContractClass, the same as the way you use into methods that are defined in types crate

Reply:
Actually I forgot to delete this TODO because I had accepted that it's working. I didn't know that you could just define something in another file and have it accessible in another file in the same crate.

Reply:
Removed TODO in https://github.com/0xSpaceShard/starknet-devnet-rs/pull/99/commits/a99fe208ab5c6fdea95edcb1d90354b9244a63e1